packages/backend-app-api/src/services/implementations/auth/external/types.ts

```diff
@@ -14,9 +14,21 @@
  * limitations under the License.
  */
 
+import { BackstagePrincipalAccessRestrictions } from '@backstage/backend-plugin-api';
 import { Config } from '@backstage/config';
 
+export type AccessRestriptionsMap = Map<
+  string, // plugin ID
+  BackstagePrincipalAccessRestrictions
+>;
+
 export interface TokenHandler {
   add(options: Config): void;
-  verifyToken(token: string): Promise<{ subject: string } | undefined>;
+  verifyToken(token: string): Promise<
+    | {
+        subject: string;
+        accessRestrictions?: AccessRestriptionsMap;
```

Comment:
Here we also already know the plugin ID right? as in this could be a `BackstagePrincipalAccessRestrictions`

Reply:
Actually no. The ExternalTokenHandler is made in `createRootContext`, so it's the `DefaultAuthService` that's responsible for picking out the right entry in the map for now. We can refactor that later though.

Reply:
Thinking it would be a bit cleaner to have the token handlers be per plugin though, it's what I was on about over here: https://github.com/backstage/backstage/pull/24321#pullrequestreview-2014475602